codegen/lib/graphql_java_gen.rb

```diff
@@ -193,12 +193,13 @@ def generate_build_input_code(expr, type, depth: 1)
       item_type = type.of_type
       <<-JAVA
         _queryBuilder.append('[');
-
-        String listSeperator#{depth} = "";
-        for (#{java_input_type(item_type)} item#{depth} : #{expr}) {
-          _queryBuilder.append(listSeperator#{depth});
-          listSeperator#{depth} = ",";
-          #{generate_build_input_code("item#{depth}", item_type, depth: depth + 1)}
+        {
+          String listSeperator#{depth} = "";
+          for (#{java_input_type(item_type)} item#{depth} : #{expr}) {
+            _queryBuilder.append(listSeperator#{depth});
+            listSeperator#{depth} = ",";
+            #{generate_build_input_code("item#{depth}", item_type, depth: depth + 1)}
+          }
```

Comment:
I don't get it. Did you just add a scope to the block?

Reply:
Yeah, it was being called again in [Input.java.erb](https://github.com/Shopify/graphql_java_gen/blob/master/codegen/lib/graphql_java_gen/templates/Input.java.erb#L93) for the optional fields but had no context to resume the previous depth count. Just added a block so it can safely start at `1` again for new fields.